### ISSUE TITLE
Staging Review finding: Important context likely to be missed by screen reader users.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#91345]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Disability Benefits Experience Team 1
**Product:** 21-526EZ
**Feature:** Toxic Exposure Page

## Findings details
**VA.gov Experience Standard - issue:** [User doesn't have enough information to complete a task.](https://design.va.gov/about/experience-standards/#user-has-enough-information-to-complete-a-task)
**VA.gov Experience Standard - category:** [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

When entering service dates for exposure, screen reader users are likely to miss the context provided to understand what they're being asked to provide.

Impacted pages:
- Post-1990 locations (eg. toxic-exposure/gulf-war-1990-location-afghanistan)
- Post-9/11 locations (eg. toxic-exposure/gulf-war-2001-location-djibouti)
- Herbicide locations (eg. toxic-exposure/herbicide-location-cambodia)
- Other exposure types (eg. toxic-exposure/additional-exposure-asbestos)

Deep into a form flow, many screen reader users don't have their screen reader announce all of the text of the page. Instead, they're trying to work through it as efficiently as possible --- for example, pressing [tab] to move through each input on the page. When tabbing to an input, the screen reader announces anything that's programmatically associated with the input: the `label`, the `legend` for the containing `fieldset`, and any text associated via `aria-describedby` or `aria-labelledby`.

As currently coded, the location (eg. "Location 1 of 17: Afghanistan")/type of hazard (eg. "Hazard 1 of 6: Asbestos") and the instructions to the user ("Enter any date ranges...") are in the `fieldset`, but being in the `fieldset` isn't enough for screen readers to consider them programmatically associated.

That means a screen reader user is likely to miss the context of what they're supposed to be entering at this page. When I tab to the first input for Afghanistan after 8/2/1990, I hear: "Service after August 2, 1990, grouping. Service start date (approximate)..." The key thing here that's missing is any mention of location, but I think the "Enter any date range..." text is important here too.

<details><summary>Link, screenshot or steps to recreate</summary>

Steps to reproduce:
1. Activate a screen reader. (Easiest testing: VoiceOver on Mac, NVDA on Windows.)
2. Navigate through the 21-526ez flow to the toxic exposure section, select a location, and press continue.
3. Press [tab] to move to the first input.
4. Note that the location information and instructions are not announced.

</details>

## Recommended action

Easiest solution here is to move anything that needs to be announced into the `legend`. Something like:
```
<fieldset>
        <legend>
                <h3 class="...">Service after August 2, 1990</h3>
                <h4 class="...">Location 1 of 17: Afghanistan</h4>
                <p class="...">Enter any date range you served in this location. You don’t need to have exact dates.</p>
        </legend>
        [... inputs...]
</fieldset>
```

You'll need to do a little extra CSS work to style the H4 and `p` so that they look the way you intend them to look. But for screen reader users, the result will be a much more helpful announcement when tabbing to that first input: "Service after August 2, 1990, Location 1 of 17: Afghanistan, Enter any date range you served in this location. You don't need to have exact dates, grouping. Service start date (approximate)..."

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by christinec-fftc** [comment]
I was able to draft a pr for this [#31576](https://github.com/department-of-veterans-affairs/vets-website/pull/31576), but I had to do an interesting workaround. Previously the title was passed into `ui:title` while the subtitle and instructions were passed into `ui:description`. In order to get them all in in the fieldset's legend together, had to put all items into `ui:title`.

**Comment by shiragoodman** [comment]
@briandeconinck FYSA
